backend/services/deviceauth/api/http/api_devauth.go

```diff
@@ -237,17 +237,14 @@ func (i *DevAuthApiHandlers) SearchDevicesV2Handler(c *gin.Context) {
 		return
 	}
 
-	numDevs := len(devs)
-	hasNext := false
-	if int64(numDevs) > perPage {
-		hasNext = true
-		numDevs = int(perPage)
-	}
+	numDevs := int64(len(devs))
+	hasNext := numDevs > perPage
+
 	hints := rest.NewPagingHints().
 		SetPage(page).
 		SetPerPage(perPage).
 		SetHasNext(hasNext).
-		SetTotalCount(int64(numDevs))
+		SetTotalCount(numDevs)
```

Comment:
This line needs to be removed as it is not the actual total count (see above). We only know if it has one device more than current page.
```suggestion
```